### PR TITLE
removed rustfilt dependency, merged all binary switches into one call. close to being one nice rust binary! 

### PR DESCRIPTION
**Pull request by laudiacay**
ignore my other PR that's a git screwup

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -2,33 +2,164 @@
...
+
+    #[structopt(long = "crate-name", value_name = "NAME")]
+    /// crate name
+    crate_name: String,
 }
 
-fn main() {
-    env_logger::init();
+/// Based on code from cargo-bloat. It seems weird that CompileOptions can be
+/// constructed without providing all standard cargo options, TODO: Open an issue
```
Comment (disconnect3d) [inline src/main.rs:87]: What do you mean by `all standard cargo options`?

File: src/matching.rs [hunk 2]
```diff
@@ -209,23 +196,36 @@ fn do_output(badness: HashMap<String, u32>) {
...
-    "$SIDEROPHILE_OUT/nodes_to_taint.txt" \
-    "$CRATENAME" \
-    > "$SIDEROPHILE_OUT/badness.txt"
-*/
-pub fn real_main(args: &TraceArgs) -> CliResult {
-    let callgraph = parse_input_data(&args.input_callgraph, &args.input_unsafe_deps_file);
-    let badness = trace_unsafety(callgraph, &args.crate_name);
+fn do_demangle(input: &PathBuf) -> io::Result<Vec<String>> {
+    let mut in_reader = BufReader::new(File::open(input)?);
+    let mangled_name_regex: Regex = Regex::new(r"_(ZN|R)[\$\._[:alnum:]]*").unwrap();
```
Comment (disconnect3d) [inline src/matching.rs:201]: we could link to https://en.wikipedia.org/wiki/Name_mangling#Rust or some better resource in a comment

File: src/matching.rs [hunk 3]
```diff
@@ -101,49 +86,46 @@ fn parse_input_data(
...
-                    match node_id_to_caller_nodes.get_mut(&cap[2].to_string()) {
-                        Some(set) => {set.insert(cap[1].to_string());},
-                        None => {
-                            let mut set : HashSet<String> = HashSet::new();
-                            set.insert((&cap[1]).to_string());
-                            node_id_to_caller_nodes.insert(cap[2].to_string(), set);}
+            }
+        } else {
+            // found a new edge!
+            for cap in edge_re.captures_iter(&contents) {
```
Comment (disconnect3d) [inline src/matching.rs:107]: not sure what cap is here completely but maybe we can unpack it like `for (_, node_id, full_label) in ...`? (same above)

File: src/main.rs (hunk 1 above)
Comment (laudiacay) [reply src/main.rs:87]: great question? that was a comment in the original code that i copypasted along with the code that i moved btwn files.

File: src/matching.rs (hunk 2 above)
Comment (laudiacay) [reply src/matching.rs:201]: added

File: src/matching.rs (hunk 3 above)
Comment (laudiacay) [reply src/matching.rs:107]: those are captures from the regex- rust doesn't support that pattern match for capture object (as far as i can tell) but if you have another idea I'm open to it!

Comment (disconnect3d) [reply src/matching.rs:107]: Ahh; not aware of a better way then :P

**Pull request merged**